x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/store/KibanaOwnedReservedRoleDescriptors.java

```diff
@@ -191,7 +191,12 @@ static RoleDescriptor kibanaSystem(String name) {
                         ".logs-osquery_manager.action.responses-*",
                         "profiling-*"
                     )
-                    .privileges(UpdateSettingsAction.NAME, PutMappingAction.NAME, RolloverAction.NAME)
+                    .privileges(
+                        UpdateSettingsAction.NAME,
+                        PutMappingAction.NAME,
+                        RolloverAction.NAME,
+                        "indices:admin/data_stream/lifecycle/put"
```

Comment:
__nit:__ Cannot you use `PutDataStreamLifecycleAction.NAME` to not hardcode the string?

Reply:
I tried to import PutDataStreamLifecycleAction here but without sucess java was not compiling

Reply:
Got it, maybe ES team can guide on that then. In any case, it's not a big deal, just an optional nit.

Reply:
The xpack core module/plugin doesn't have direct visibility to the dlm module/plugin which is why you can't reference the name directly. I would have a minor preference to put the action name in a variable in [IndexPrivilege.java](https://github.com/elastic/elasticsearch/blob/main/x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/privilege/IndexPrivilege.java) so that it is more centralized. 

The core the change here (adding the priv) seems fine to me but  @n1v0lg - can you take a look too, i don't remember the details of the this feature you assisted with.

Reply:
If `.logs-endpoint.action.responses-*` and the other indices with the leading dot are **not** system indices/system data streams, this change is fine. Otherwise, we'd have to update the internal user privileges here as well: https://github.com/elastic/elasticsearch/blob/main/x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/user/InternalUsers.java#L133  

@nchaulet could you confirm those are not system indices/system data streams? I don't think they are since the privilege definition does not include `allowRestrictedIndices` but just double-checking. 

As far the privilege itself goes, `"indices:admin/data_stream/lifecycle/put"` is fine -- there is also a named privilege `manage_data_stream_lifecycle` which grants access to the PUT, GET, and DELETE data stream lifecycle APIs. If only updating is required though, going with the specific action as you do works and is good in terms of least privilege.

Reply:
the `.logs-*` are not system indices just some hidden datastreams used by elastic agent and defined in integrations.

It seems we only need the PUT currently